### DISCUSSION TITLE
on notification click

### DISCUSSION THREAD

**Discussion by link-leo** [felixwiemuth/SimpleReminder#46] [Ideas and feedback]
I'd like to see the option to add an event to clicking the notification. Something like call/sms # (which can be entered as URIs like `tel/sms:0000000000`, go to URLs, open selected app, trigger a [Tasker](https://tasker.joaoapps.com/) task as [HTTP-Shortcuts does](https://github.com/Waboodoo/HTTP-Shortcuts/blob/develop/docs/scripting.md#trigger-tasker-task), or (for the advanced users) send an intent.

I absolutely *love* the simplified aesthetic/use of this app and I don't want to take away from that, so maybe the fields to select/configure these options could be hidden by default but enabled via the settings menu or something.

Anyway, **love** this app!

**Comment by felixwiemuth** [comment]
Hi,

Thanks for the great suggestion!

Related to this, it is already the plan that more actions can be executed from a notification:
A notification has several controls (like tap, long-tap, buttons, swipe) and actions that can be assigned to these controls (like edit the reminder, share the reminder's text etc.).

Your suggestion could be integrated into this by allowing a "reminder action", which is a user-defined action added when creating the reminder, like "call this number". So in settings, one could for example define that "tap on the reminder" executes the reminder action.

Alternatively, and maybe to start with, the notification could simply get an action button which executes the reminder action (if the reminder has one).

Regarding defining the actions, while a specific UI to do this will be nice to have, I would start with a text-based format, added as part of the reminder text. If an action is recognized in the text, when the reminder is due it will be removed from the text and added as an action to the notification.

The format could be a custom format, or directly Android Intent URIs (which would be even easier to implement). However, here security issues must be considered, especially if at some point we allow to share/receive reminders from others. It must be ensured that the user does not execute actions they don't intend to execute.

Finally, I agree that the app should always stay easy to use. Thus, even though many interesting and more or less complex features are planned, they must be added in a way so they are not in the way when you don't need them.

And thanks for the nice words, very much appreciated!

**Reply by link-leo** [reply]
That sounds really excellent. I'm glad to hear you took it in and liked it. I've also been thinking about this in the way [ntfy](https://github.com/binwiederhier/ntfy) does things. You can easily add [actions](https://docs.ntfy.sh/publish/#action-buttons) to either the [click](https://docs.ntfy.sh/publish/#click-action) event or the pull-down buttons in a spawned notification, including [broadcasting an intent](https://docs.ntfy.sh/publish/#send-android-broadcast) or [sending an http request](https://docs.ntfy.sh/publish/#send-http-request). Something like this would be great.
